### PR TITLE
Cleanup code

### PR DESCRIPTION
**Pull request by CarsonF**
:see_no_evil:

### CODE REVIEW

File: src/Content.php [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 namespace Bolt;
 
+use Maid\Maid;
 use Silex;
 use Symfony\Component\Filesystem\Filesystem;
 use Bolt\Library as Lib;
```
Comment (GawainLynch) [inline src/Content.php]: OCD fail :-P

Comment (CarsonF) [inline src/Content.php]: That they are the same?

File: src/Library.php [hunk 2]
```diff
@@ -169,11 +172,11 @@ public static function simpleredirect($path, $abort = false)
...
         echo "<p>Redirecting to <a href='$path'>$path</a>.</p>";
         echo "<script>window.setTimeout(function () { window.location='$path'; }, 500);</script>";
-        if ($abort) {
-            return $app->abort(303, "Redirecting to '$path'.");
+        if (!$abort) {
+            return $path;
         }
 
-        return $path;
+        $app->abort(303, "Redirecting to '$path'.");
```
Comment (GawainLynch) [inline src/Library.php]: One step further:
```php
$app->abort(Response::HTTP_SEE_OTHER, "Redirecting to '$path'.");
```

File: src/Content.php (hunk 1 above)
Comment (GawainLynch) [inline src/Content.php]: Alphabetical order on the use statements… Last time I checked 'B' came before 'M' and 'S'.

**Comment by GawainLynch** [comment]
Travis barfing:
```
1) Bolt\Tests\Logger\ChangeLogTest::testGetChangelog
Doctrine\DBAL\Exception\TableNotFoundException: An exception occurred while executing 'SELECT * FROM log_change ORDER BY id ASC LIMIT 1 OFFSET 0':
SQLSTATE[HY000]: General error: 1 no such table: log_change
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractSQLiteDriver.php:58
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:833
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php:206
/home/travis/build/bolt/bolt/src/Logger/ChangeLog.php:55
/home/travis/build/bolt/bolt/tests/Logger/ChangeLogTest.php:31
```

**Comment by CarsonF** [comment]
So that's not my fault?

File: src/Composer/Action/UpdatePackage.php [hunk 3]
```diff
@@ -32,17 +31,22 @@ public function __construct(Application $app)
...
         $composer = $this->app['extend.manager']->getComposer();
         $io = $this->app['extend.manager']->getIO();
-        $options = $this->app['extend.manager']->getOptions();
+        $packageManagerOptions = $this->app['extend.manager']->getOptions();
 
         // Handle passed in options
-        if (!is_null($options)) {
-            $options = Arr::mergeRecursiveDistinct($options, $options);
+        if (!$options) {
+            $options = array_replace_recursive($packageManagerOptions, $options);
```
Comment (GawainLynch) [inline src/Composer/Action/UpdatePackage.php]: This will break some keys

Comment (CarsonF) [inline src/Composer/Action/UpdatePackage.php]: What keys??

Comment (GawainLynch) [inline src/Composer/Action/UpdatePackage.php]: `array_replace_recursive()` will overwrite/remove stuff in multidimensional arrays.  We had a really bad case of this in 2.0-dev that took us ages to track down.

Comment (CarsonF) [inline src/Composer/Action/UpdatePackage.php]: Really? I didn't know that...I'll revert. What's an example?

Comment (GawainLynch) [inline src/Composer/Action/UpdatePackage.php]: Pseudo example:
```php
$a = array(
    'foo' => array(
        'cow' => 'moo',
        'dog' => 'whoof'
    )
);

$b = array(
    'foo' => array(
        'cow' => 'steak'
    )
);
```

Merging those would wipe the 'dog' key.  You might need to go one deeper, can't remember.

**Comment by GawainLynch** [comment]
> So that's not my fault?

I don't know, PR before this passed. This one failed… :grinning:

File: src/Composer/Action/UpdatePackage.php (hunk 3 above)
Comment (CarsonF) [inline src/Composer/Action/UpdatePackage.php]: are you sure you weren't using `array_replace`?

Comment (GawainLynch) [inline src/Composer/Action/UpdatePackage.php]: Positive

File: src/Composer/Action/RequirePackage.php [hunk 4]
```diff
@@ -69,6 +77,7 @@ public function execute(array $package)
         }
 
         // Get the Composer repos
+        // TODO What is repos used for?
         $repos = $composer->getRepositoryManager()->getRepositories();
```
Comment (GawainLynch) [inline src/Composer/Action/RequirePackage.php]: This is from Composer's implementation of the class. I didn't want to remove it until I had a chance to talk to Jordi about it.

File: src/Composer/Action/RequirePackage.php [hunk 5]
```diff
@@ -119,6 +128,7 @@ public function execute(array $package)
             if ($status !== 0) {
                 if ($newlyCreated) {
                     // Installation failed, deleting JSON
+                    // TODO json isn't defined
```
Comment (GawainLynch) [inline src/Composer/Action/RequirePackage.php]: Ugh, something I haven't finished properly. Log the issue please.

File: src/Composer/Action/SearchPackage.php [hunk 6]
```diff
@@ -54,6 +57,7 @@ public function execute($packages)
             $repos = new CompositeRepository(array_merge(array($installedRepo), $defaultRepos));
         }
 
+        // TODO onlyname doesn't exist
```
Comment (GawainLynch) [inline src/Composer/Action/SearchPackage.php]: Log please

File: src/Composer/Action/RequirePackage.php (hunk 4 above)
Comment (CarsonF) [inline src/Composer/Action/RequirePackage.php]: Looks like PlatformRepository is used for determining requirements like "ext-*", php, etc. But it is never getting add to the Pool given to the VersionSelector.

File: src/Configuration/ResourceManager.php [hunk 7]
```diff
@@ -41,12 +40,16 @@ class ResourceManager
 
     protected $urls = array();
 
+    /** @var array TODO do we really need this? */
     protected $request = array();
```
Comment (GawainLynch) [inline src/Configuration/ResourceManager.php]:
```php
    public function setRequest($name, $value)
    {
        $this->request[$name] = $value;
    }

    public function getRequest($name)
    {
        if (! array_key_exists($name, $this->request)) {
            throw new \InvalidArgumentException("Request componenet $name is not available", 1);
        }

        return $this->request[$name];
    }
```

Comment (CarsonF) [inline src/Configuration/ResourceManager.php]: The only usages of those methods are inside the class

File: src/Logger/ChangeLog.php [hunk 8]
```diff
@@ -28,7 +29,7 @@ public function __construct(Application $app)
     {
         $this->app = $app;
 
-        $prefix = $app['config']->get('general/database/prefix', "bolt_");
+        $prefix = $app['config']->get('general/database/prefix');
```
Comment (GawainLynch) [inline src/Logger/ChangeLog.php]: This is what is breaking Travis. For some reason nothing in getting returned.

Comment (CarsonF) [inline src/Logger/ChangeLog.php]: Latest commit should fix it

File: src/Composer/Action/RequirePackage.php (hunk 4 above)
Comment (GawainLynch) [inline src/Composer/Action/RequirePackage.php]: It's not costing us anything, let's just leave it for now and axe the TODO :-P

File: src/Logger/ChangeLog.php (hunk 8 above)
Comment (GawainLynch) [inline src/Logger/ChangeLog.php]: Still no :crying_cat_face: 
```
1) Bolt\Tests\Logger\ChangeLogTest::testGetChangelogEntry
Doctrine\DBAL\Exception\SyntaxErrorException: An exception occurred while executing 'SELECT log.*p FROM bolt_log_change log LEFT JOIN bolt_pages content ON content.id = log.contentid WHERE (log.id = ?) AND (log.contentid = ?) AND (contenttype = ?) LIMIT 1' with params [1, 1, "pages"]:
SQLSTATE[HY000]: General error: 1 near "p": syntax error
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractSQLiteDriver.php:74
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:825
/home/travis/build/bolt/bolt/vendor/doctrine/dbal/lib/Doctrine/DBAL/Query/QueryBuilder.php:206
/home/travis/build/bolt/bolt/src/Logger/ChangeLog.php:303
/home/travis/build/bolt/bolt/src/Logger/ChangeLog.php:161
/home/travis/build/bolt/bolt/tests/Logger/ChangeLogTest.php:77
```

**Comment by GawainLynch** [comment]
6ba23bef968df038861c10d827957314c10d083f seems to have not been liked by Travis :gun:

**Comment by CarsonF** [comment]
I think this is ready to go now!

**Comment by bobdenotter** [comment]
All aboard the merge train!

**Pull request merged**